config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,71 @@
+class Rack::Attack 
+  LOGIN_ROUTES = %w[/api/auth /dashboard/authenticate].freeze
+
+  default_limit  = ::Settings.rate_limiting.default.limit
+  default_period = ::Settings.rate_limiting.default.period
+  login_limit    = ::Settings.rate_limiting.login.limit
+  login_period   = ::Settings.rate_limiting.login.period
```

Comment:
1. These are evaluated at load time, so if the user change it they won't be applied.  They would have to restart the UI workers.  Is it possible to evaluate them at run time?
2. These are local variables - is that intentional?  I see their usage as part of the DSL calls below, but wasn't sure if that was intentional, or if this should be setting some class variables or something instead.

Reply:
> # You can also set a limit and period using a proc. For instance, after
> # Rack::Auth::Basic has authenticated the user:
> limit_proc = proc { |req| req.env["REMOTE_USER"] == "admin" ? 100 : 1 }
> period_proc = proc { |req| req.env["REMOTE_USER"] == "admin" ? 1 : 60 }

Not sure when the proc gets evaluated, but it seems we can use procs instead, and that can give us the "dynamic" throttling

Reply:
> These are local variables - is that intentional?

Yes, just for readability, so the `.trottle` lines weren't so long.   It is within the scope of the `class` definitions, so it works just fine, but doesn't make it accessible outside of when you are re-opening the class.

Should work the same when we don't open up the `Rack::Attack` class directly.

Reply:
Fixed.